sql/driver/operators_it_test.go

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package driver_test
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	hz "github.com/hazelcast/hazelcast-go-client"
+	"github.com/hazelcast/hazelcast-go-client/internal/it"
+	"github.com/hazelcast/hazelcast-go-client/serialization"
+	"github.com/hazelcast/hazelcast-go-client/sql/driver"
+)
+
+type employee struct {
+	PersonnelInfo `json:"personnel_info"`
+	CompanyInfo   `json:"company_info"`
+	ID            int `json:"id"`
+}
+type PersonnelInfo struct {
+	Name string `json:"name"`
+	Age  int    `json:"age"`
+}
+type CompanyInfo struct {
+	Title   string `json:"title"`
+	DepCode int    `json:"dep_code"`
+}
+
+func TestNestedJSONQuery(t *testing.T) {
+	it.SkipIf(t, "hz < 5.1")
+	it.SQLTester(t, func(t *testing.T, client *hz.Client, config *hz.Config, m *hz.Map, mapName string) {
+		defer driver.SetSerializationConfig(nil)
+		const rowCount = 50
+		db := initDB(t, config)
+		defer db.Close()
+		ms := createMappingStr(mapName, "bigint", "json")
+		it.Must(createMapping(t, db, ms))
+		// Insert random employee entries.
+		employees, err := populateMapWithEmployees(m, rowCount)
+		it.Must(err)
+		// Test query on nested json
+		complexQuery := fmt.Sprintf(`SELECT JSON_VALUE(this, '$.personnel_info.name') FROM "%s" WHERE CAST(JSON_VALUE(this, '$.id') AS TINYINT)=0`, mapName)
+		row := db.QueryRowContext(context.Background(), complexQuery)
+		var name string
+		it.Must(row.Scan(&name))
+		assert.Equal(t, employees[0].PersonnelInfo.Name, name)
+	})
+}
+
+func TestJSONOperators(t *testing.T) {
+	it.SkipIf(t, "hz < 5.1")
+	it.SQLTester(t, func(t *testing.T, client *hz.Client, config *hz.Config, m *hz.Map, mapName string) {
+		const rowCount = 50
+		defer driver.SetSerializationConfig(nil)
+		db := initDB(t, config)
+		defer db.Close()
+		ms := createMappingStr(mapName, "bigint", "json")
+		it.Must(createMapping(t, db, ms))
+		// Create a view to test.
+		createViewQuery := fmt.Sprintf(`CREATE VIEW Personnel AS SELECT CAST(JSON_VALUE(this, '$.id') AS int) as pid, JSON_QUERY(this, '$.personnel_info') as info FROM "%s"`, mapName)
```

Comment:
Where does `Create view` statement come from? I could not find a doc?

Reply:
@emreyigit It's in the beta doc: https://docs.hazelcast.com/hazelcast/5.1-beta-1/sql/create-view